hazelcast/src/main/java/com/hazelcast/instance/DefaultNodeContext.java

```diff
@@ -76,63 +75,58 @@ public AddressPicker createAddressPicker(Node node) {
         Properties properties = memberAddressProviderConfig.getProperties();
-        MemberAddressProvider memberAddressProvider = newMemberAddressProviderInstance(constructor, properties);
+        MemberAddressProvider memberAddressProvider = newMemberAddressProviderInstance(clazz,
+                memberAddressProviderLogger, properties);
         return new DelegatingAddressPicker(memberAddressProvider, config.getNetworkConfig(), addressPickerLogger);
-
     }
 
-    private MemberAddressProvider newMemberAddressProviderInstance(Constructor<?> constructor, Properties properties) {
-        Class<?>[] parameterTypes = constructor.getParameterTypes();
-        Class<?> clazz = constructor.getDeclaringClass();
-        String classname = clazz.getName();
-        try {
-            if (parameterTypes.length == 0) {
-                //we have only the no-arg constructor -> we have to fail-fast when some properties were configured
-                if (properties != null && !properties.isEmpty()) {
-                    throw new ConfigurationException("Cannot find a matching constructor for MemberAddressProvider.  "
-                            + "The member address provider has properties configured, but the class " + "'" + classname
-                            + "' does not have a public constructor accepting properties.");
-                }
-
-                return (MemberAddressProvider) constructor.newInstance();
-            } else {
-                if (properties == null) {
-                    properties = new Properties();
-                }
-                return (MemberAddressProvider) constructor.newInstance(properties);
-            }
-        } catch (InstantiationException e) {
-            throw new ConfigurationException("Cannot create a new instance of MemberAddressProvider '" + clazz + "'", e);
-        } catch (IllegalAccessException e) {
-            throw new ConfigurationException("Cannot create a new instance of MemberAddressProvider '" + clazz + "'", e);
-        } catch (InvocationTargetException e) {
-            throw new ConfigurationException("Cannot create a new instance of MemberAddressProvider '" + clazz + "'", e);
+    @SuppressWarnings("checkstyle:npathcomplexity")
+    // justification for the suppression: the flow is pretty straightforward and breaking this up into smaller pieces
+    // would make it harder to read
+    private static MemberAddressProvider newMemberAddressProviderInstance(Class<? extends MemberAddressProvider> clazz,
+                                                                          ILogger logger, Properties properties) {
+        Properties nonNullProps = properties == null ? new Properties() : properties;
+        MemberAddressProvider provider = InstantiationUtils.newInstanceOrNull(clazz, nonNullProps, logger);
+        if (provider == null) {
+            provider = InstantiationUtils.newInstanceOrNull(clazz, logger, nonNullProps);
```

Comment:
Not easier on the eyes to do a static import? Long method suck, but wide methods suck equally much imho.

Reply:
I tend to use static imports for well-known classes. here, it's questionable. I dont have strong opinion tho.

Reply:
I use static imports as long as it doesn't sacrifice readability. I find long lines equally annoying as long methods.